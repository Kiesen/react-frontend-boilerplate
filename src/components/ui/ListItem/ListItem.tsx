import React, { FC } from 'react';
import { useTranslation } from 'react-i18next';

import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

// -- TODO --
// To make the ListItem component more generic it is possible
// to specify explicitly some default properties.
interface Props {
  item: Record<string, string | number | boolean>;
}

/**
 * A simple ui component which renders the content
 * of a given item. Uses material ui card components as
 * wrapper for the styling.
 *
 * @param {Object} item - an item object with userId, title and body
 * @returns a simple list item using material ui card components
 */
const ListItem: FC<Props> = ({ item }) => {
  // i18n translations
  const { t } = useTranslation();

  return (
    <Card>
      <CardContent>
        <Typography color="textSecondary" gutterBottom>
          {t('client:listItem.user')}{' '}
          {item.userId ? item.userId : t('client:listItem.anonymous')}
        </Typography>
        <Typography variant="h5" component="h3">
          {item.title ? item.title : ''}
        </Typography>
        <Typography variant="body2" color="textSecondary" component="p">
          {item.body ? item.body : ''}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default ListItem;
