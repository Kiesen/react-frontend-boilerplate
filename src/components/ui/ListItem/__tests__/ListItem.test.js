import React from 'react';
import { shallow } from 'enzyme';

import ListItem from 'components/ui/ListItem/ListItem';

describe('<ListItem>', () => {
  // ListItem has item as a required prop!
  const item = {};

  it('renders correctly', () => {
    const wrapper = shallow(<ListItem item={item} />);
    expect(wrapper).toMatchSnapshot();
  });
});
