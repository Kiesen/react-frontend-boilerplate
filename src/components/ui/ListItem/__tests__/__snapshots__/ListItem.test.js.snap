// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem> renders correctly 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
      gutterBottom={true}
    >
      client:listItem.user
       
      client:listItem.anonymous
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      component="h3"
      variant="h5"
    />
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
      component="p"
      variant="body2"
    />
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
